### DISCUSSION TITLE
v3_swap_exact_in by default communicates to Permit2

### DISCUSSION THREAD

**Discussion by Nikotos** [Elnaril/uniswap-universal-router-decoder#60] [Q&A]
Hey!

I've been actively using your library and haven't faced any issues so far. 

However, there is one small bug, which complicates things a bit

```
encoded_input = (
            codec
            .encode
            .chain()
            .v3_swap_exact_in(FunctionRecipient.ROUTER, tokens_amount, eth_amount_out_min, sell_path)
            .unwrap_weth(FunctionRecipient.SENDER, eth_amount_out_min)
            .build(int(time.time() + deadline_seconds))
    )
```

in the code above the resulting transaction tries to withdraw tokens from tx sender via Permit2 patters (however, the permit2 patters is not instructed here explicitly)

<img width="1095" alt="Screenshot 2024-05-11 at 5 29 29 PM" src="https://github.com/Elnaril/uniswap-universal-router-decoder/assets/25106067/476f38fe-a478-4108-a14b-312f51fb9100">

Previously I've been using pattern with Permit2 signature and everything worked perfectly.

**However, it seems that direct approval option is not working**

**Comment by Elnaril** [comment]
Hello @Nikotos !

> However, it seems that direct approval option is not working

How do  you do the direct approval? In particular, what contracts do you approve and how? Could you please share the approval part of your code?
